src/replication/adapter/adapter.go

```diff
@@ -31,9 +31,13 @@ const (
 )
 
 var registry = map[model.RegistryType]Factory{}
+var adapterInfoMap = map[model.RegistryType]*model.AdapterPattern{}
 
 // Factory creates a specific Adapter according to the params
-type Factory func(*model.Registry) (Adapter, error)
+type Factory interface {
```

Comment:
I still has some confusion about this change here. Let's have more discussions offline. @bitsf @ywk253100